flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/BufferProvider.java

```diff
@@ -42,15 +42,15 @@
 	 *
 	 * <p>Returns <code>null</code> if no buffer is available or the buffer provider has been destroyed.
 	 */
-	BufferBuilder requestBufferBuilder() throws IOException;
+	BufferBuilder requestBufferBuilder(int targetChannel) throws IOException;
```

Comment:
I guess that if instead we would overload those methods, and keep both versions: with and without `targetChannel` parameter, that would reduce the amount of changes in the code AND it would prevent from `UNKNOWN_CHANNEL` leaking to the API of `BufferPool` class - `UNKNOWN_CHANNEL` could then be a private implementation detail of the `LocalBufferPool` class.